synapse/api/errors.py

```diff
@@ -145,6 +145,22 @@ def error_dict(self):
         return cs_error(self.msg, self.errcode)
 
 
+class ShadowBanError(SynapseError):
+    """An exception that returns a 200 OK with a fake body.
+    """
+
+    def __init__(self, body=None):
+        if body is None:
+            body = {}
+
+        self.body = body
+
+        super().__init__(200, "")
```

Comment:
any reason not to do this as the first thing in the method? convention is to call the superclass constructor first.